### DISCUSSION TITLE
Dynamic routes and build-time data loading

### DISCUSSION THREAD

**Discussion by pavelhoral** [vuejs/vitepress#2711] [Q&A]
Hello, I have quite straight forward use-case that I am not sure if Vitepress supports:

* I have a large number of big JSON files. I want for each file to generate single page (e.g. `data/1.html`, `data/2.html`, ...).
* I am able to do dynamic routing with `data/[id].md` and `data/[id].paths.ts` - this is quite simple as it only extracts IDs from filenames.
* I want to have JSON content of data files statically imported in every generated page. I don't want to deploy those JSON files alongside my website and load them dynamically.
* I am not sure how to load a single file based on dynamic route from the page template.

Thank you for any suggestion.

**Comment by brc-dd** [comment]
https://vitepress.dev/guide/data-loading ?

**Reply by pavelhoral** [reply]
But is it possible to load multiple resources with that? I don't see a way to do that.

**Reply by pavelhoral** [reply]
To be more specific - I don't want to load everything in one go and have it embedded in every page. I want to load single file for a single page... something like `[id].data.ts`, but that is AFAIK not possible.

**Reply by brc-dd** [reply]
Ah, yeah. That's not possible at the moment. I haven't tried but some hack like `params: { id: ..., data: JSON.stringify(corresponding_json) }` with `const data = JSON.parse(params.value.data)` might work.

**Reply by pavelhoral** [reply]
Ok. Thank you for the confirmation. I might try creating my own simple plugin inspired by staticDataPlugin then (last words before going insane :))...
